six/modules/c++/samples/check_valid_six.cpp

```diff
@@ -148,16 +148,26 @@ int main(int argc, char** argv)
             log->info(Ctxt("Reading " + inputPathname));
             try
             {
-                if (nitf::Reader::getNITFVersion(inputPathname) ==
-                        NITF_VER_UNKNOWN)
+                std::auto_ptr<six::Data> data =
+                        six::parseDataFromFile(xmlRegistry,
+                                               inputPathname,
+                                               schemaPaths,
+                                               *log);
+
+                six::sicd::ComplexData* complexData =
```

Comment:
Rather than a `dynamic_cast`, you can use `six::Data::getDataType()` and see if it is complex.  If it is, you can use `reinterpret_cast`.  This is cheaper (`dynamic_cast` needs to do an expensive check) - not that performance matters for this - and is consistent with how other parts of SIX do this check (which admittedly I have mixed feelings about).